doc/user-guide/Jingle-SIP-setup.md

```diff
@@ -0,0 +1,300 @@
+    * see [How-to-build](How-to-build.md) for details on building.
+      It's important to remember to run the configuration script with `with-jingle-sip` flag set:
+      `tools/configure with-jingle-sip`.
+      Without this, 3rd party deps required by the Jingle/SIP translator will not be included in the release.
+
+* Routr (SIP server) - https://routr.io
+    * I recommend downloading binaries for your system from https://routr.io/docs/getting-started-installation.html
+* Jitsi (XMPP and SIP client application) - https://desktop.jitsi.org/
+* Otalk - web based XMPP client - https://github.com/otalk/otalk-im-client
+    * Folow the instructions on https://github.com/otalk/otalk-im-client#installing to run it
+
+We will use 2 users `xmpp.user@xmpp.example` and `sip.user@sip.example`.
+
+### Configuring Routr
+
+First the domain `sip.example` needs to added to domains served by Routr.
+To do it, paste the following content to config/domains.yml in the directory where Routr was:
+
+```yml
+- apiVersion: v1beta1
+  kind: Domain
+  metadata:
+    name: SIP domain
+  spec:
+    context:
+      domainUri: sip.example
+```
+
+Then the `sip.user@sip.example` needs to be added to `config/agents.yml` like below:
+
+```yml
+- apiVersion: v1beta1
+  kind: Agent
+  metadata:
+    name: SIP User
+  spec:
+    credentials:
+      username: 'sip.user'
+      secret: '1234'
+    domains: [sip.example]
+```
+
+Now Routr can be started with
+
+    ./routr
+
+If all goes well we'll see the following output:
+
+```
+[INFO ] Starting Routr
+[INFO ] Listening  on 10.152.1.27:5060 [udp]
+[INFO ] Listening  on 10.152.1.27:5060 [tcp]
+[INFO ] Starting Location service
+[INFO ] Starting Registry service
+[INFO ] Starting Restful service (port: 4567, apiPath: '/api/v1beta1')
+```
+
+It is important to remember the IP address as it'll be used in next point.
+
+#### A side not
+
+In Routr's logs you may see messages like
+
+    [WARN ] Unable to register with Gateway -> sip.provider.net. (Verify your network status)
+
+or
+
+    [ERROR] java.lang.RuntimeException: javax.sip.header.TooManyHopsException: has already reached 0!
+
+They can be ignored for the purpose of the tutorial.
+
+### Configuring /etc/hosts
+
+In my case the IP reported by Routr was `10.152.1.27`.
+Now we need to use this to update `/etc/hosts` file like below:
+
+    10.152.1.27     sip.example xmpp.example
+
+### Configuring MongooseIM
+
+At this point I assume that MongooseIM was built with `make rel`, that it is running and the CWD is `_build/prod/rel/mongooseim`.
+Similar to Routr, MongooseIM also needs to know which hosts to server.
+Please replace the default host defined in file `etc/ejabberd.cfg` in line
+
+```erlang
+{hosts, ["localhost"] }.
+```
+
+to:
+
+```erlang
+{hosts, ["xmpp.example", "sip.example"] }.
+```
+
+Now we need to enable `mod_jingle_sip`, please add the following line in modules list (somewhere around line 740 in the same file)
```

Comment:
line number can change, doesn't it? Will we remember to update it?

Reply:
That's why I said `around` and above pointed to the doc describing how to configure modules.